amespahdbpythonsuite/amespahdb.py

```diff
@@ -0,0 +1,707 @@
+                                  'description': ''},
+                           units={'abscissa': {'unit': 1,
+                                               'str': 'frequency [wavenumber]'},
+                                  'ordinate': {'unit': 2,
+                                               'str': 'integrated cross-section [km/mol]'}}
+                           )
+
+    def getlaboratorybyuid(self, uids):
+        """
+        Retrieve and return laboratory database object based on UIDs input.
+        UIDs can be a list, e.g. the output of search method, or an integer.
+        Calls the :meth:`amespahdbpythonsuite.laboratory.laboratory` class.
+
+        Parameters
+        ----------
+        uids : list of integers
+            List of UIDs.
+
+        Returns
+        -------
+        laboratory database object
+
+        """
+        # Check if the experimental database is loaded.
+        if self.__data['type'] != 'experimental':
+
+            self.message('EXPERIMENTAL DATABASE REQUIRED')
+
+            return None
+
+        from amespahdbpythonsuite.laboratory import laboratory
+
+        if type(uids) == int:
+            uids = [uids]
+
+        return laboratory(type=self.__data['type'],
+                          version=self.__data['version'],
+                          data=self.__getkeybyuids('laboratory', uids),
+                          pahdb=self.__data,
+                          uids=uids,
+                          model={'type': 'laboratory_m',
+                                 'temperature': 0.0,
+                                 'description': ''},
+                          units={'abscissa': {'unit': 1,
+                                              'str': 'frequency [wavenumber]'},
+                                 'ordinate': {'unit': 2,
+                                              'str': 'absorbance [-log(I/I$_{0})$'}}
+                          )
+
+    def getspeciesbyuid(self, uids):
+        """
+        Retrieve and return species object based on UIDs input.
+        UIDs can be a list, e.g. the output of search method, or an integer.
+        Calls the :meth:`amespahdbpythonsuite.species.species` class.
+
+        Parameters
+        ----------
+        uids : list of integers
+            List of UIDs.
+
+        Returns
+        -------
+        species object
+
+        """
+        from amespahdbpythonsuite.species import species
+
+        if type(uids) == int:
+            uids = [uids]
+
+        return species(type=self.__data['type'],
+                       version=self.__data['version'],
+                       data=self.__getkeybyuids('species', uids),
+                       pahdb=self.__data,
+                       uids=uids,
+                       references=self.__getkeybyuids('references', uids),
+                       comments=self.__getkeybyuids('comments', uids)
+                       )
+
+    def getgeometrybyuid(self, uids):
```

Comment:
It is probably best to implement a 'mock' geometry-class so that the Travis build doesn't fail.